qa/suites/orch/cephadm/workunits/task/test_monitoring_stack_basic.yaml

```diff
@@ -0,0 +1,52 @@
+roles:
+- - host.a
+  - mon.a
+  - mgr.a
+- - host.b
+  - mon.b
+  - mgr.b
+- - host.c
+  - mon.c
+tasks:
+- install:
+- cephadm:
+- cephadm.shell:
+    host.a:
+      - |
+        set -e
+        set -x
+        ceph orch apply node-exporter
+        ceph orch apply grafana
+        ceph orch apply alertmanager
+        ceph orch apply prometheus
+        sleep 240
+        ceph orch ls
+        ceph orch ps
+        ceph orch host ls
+        MON_DAEMON=$(ceph orch ps --daemon-type mon -f json | jq -r 'last | .daemon_name')
+        GRAFANA_HOST=$(ceph orch ps --daemon-type grafana -f json | jq -e '.[]' | jq -r '.hostname')
+        PROM_HOST=$(ceph orch ps --daemon-type prometheus -f json | jq -e '.[]' | jq -r '.hostname')
+        ALERTM_HOST=$(ceph orch ps --daemon-type alertmanager -f json | jq -e '.[]' | jq -r '.hostname')
+        GRAFANA_IP=$(ceph orch host ls -f json | jq -r --arg GRAFANA_HOST "$GRAFANA_HOST" '.[] | select(.hostname==$GRAFANA_HOST) | .addr')
+        PROM_IP=$(ceph orch host ls -f json | jq -r --arg PROM_HOST "$PROM_HOST" '.[] | select(.hostname==$PROM_HOST) | .addr')
+        ALERTM_IP=$(ceph orch host ls -f json | jq -r --arg ALERTM_HOST "$ALERTM_HOST" '.[] | select(.hostname==$ALERTM_HOST) | .addr')
+        # check each host node-exporter metrics endpoint is responsive
+        ALL_HOST_IPS=$(ceph orch host ls -f json | jq -r '.[] | .addr')
+        for ip in $ALL_HOST_IPS; do
+          curl -s http://${ip}:9100/metric
+        done
+        # check grafana endpoints are responsive and database health is okay
+        curl -k -s https://${GRAFANA_IP}:3000/api/health
+        curl -k -s https://${GRAFANA_IP}:3000/api/health | jq -e '.database == "ok"'
```

Comment:
I'd add some extra checks that check that all pieces fit together:
```bash
# Validate the data source/s:
$PROMETHEUS_URL="http://prometheus:9090"
curl -k https://admin:admin@${GRAFANA_IP}:3000/api/datasources | jq --arg PROMETHEUS_URL "$PROMETHEUS_URL" -e '.[] | select(.url == $PROMETHEUS_URL )'

# Validate that Grafana Dashboards for Ceph are correctly loaded (this could be replaced with the "uid": "BGnzPcNVz", since it's unlikely to change):

curl -k https://admin:admin@${GRAFANA_IP}:3000/api/search?query=% | jq -e '.[] | select( .title == "Ceph - Cluster")'
```